### DISCUSSION TITLE
OAuth custom params

### DISCUSSION THREAD

**Discussion by beniw** [directus/directus#9691] [Archive]
Hello everyone

Did the ENV variable which before was OAUTH_{PROVIDER}_CUSTOM_PARAMS change to something else in 9.0.0?

We used it before to send a custom parameter to our OAuth provider like this:

`OAUTH_{PROVIDER}_CUSTOM_PARAMS: "{\"param\": \"value\"}"`

I'm asking here because I could not really find something in the documentation.

Many thanks and kind regards

**Comment by rijkvanzanten** [comment]
cc @aidenfoxx

**Comment by aidenfoxx** [comment]
It was a grant feature that wasn't re-implemented when we refactored the OAuth flows. It wouldn't be hard to implement since it's just passing in additional variables to the `authorizationUrl` function.

**Reply by rijkvanzanten** [reply]
@beniw Wanna try your hand at a PR to implement that? 🙂

**Comment by nicam** [comment]
It seems this issue has been resolved with the following Pull Request: https://github.com/directus/directus/pull/10301/files

The env variable is now called: `OAUTH_{PROVIDER}_PARAMS: "{\"param\": \"value\"}"`

The documentation doesn't show this yet.

**Reply by aidenfoxx** [reply]
@rijkvanzanten Is that actually valid? I thought it would always require `OAUTH_<provider>_PARAMS__<param>="Hello"`.

Either way, I didn't document it because I've learnt that my documentation causes more confusion that solutions. 😂

**Reply by rijkvanzanten** [reply]
@aidenfoxx Both are valid! 

`AUTH_GOOGLE_PARAMS__HELLO="world"`

and

`AUTH_GOOGLE_PARAMS="{\"hello\": \"world\"}"`

result in the same thing 👍🏻

**Reply by rijkvanzanten** [reply]
Added it to the docs in 0cfb901b0

**Reply by nicam** [reply]
🙏 very nice thank you!

**Reply by chrisbartley** [reply]
For people using JavaScript config files, note that you can specify params like this:

```JavaScript
AUTH_GOOGLE_PARAMS : {hd: 'example.com'}
```

(That particular example is for the [optional `hd` param](https://developers.google.com/identity/protocols/oauth2/openid-connect#hd-param) for OAuth2 authentication.)
